### PR TITLE
tests: enable a lot of the tests of main on uc20

### PR DESCRIPTION
**Pull request by mvo5**
A bunch of tests are still disabled. The ones that need attention have a `TODO:UC20:` with an explanation what needs to happen.

### CODE REVIEW

File: tests/main/install-store/task.yaml [hunk 1]
```diff
@@ -43,5 +43,12 @@ execute: |
...
     snap install --edge test-snapd-complexion
 
-    echo "All snap blobs are 0600"
-    test "$( find /var/lib/snapd/{snaps,cache,seed/snaps}/ -type f -printf '%#m\n' | sort -u | xargs )" = "0600"
+    # shellcheck source=tests/lib/systems.sh
+    . "$TESTSLIB/systems.sh"
+    # TODO:UC20: fix to work on uc20 too
+    # The "seed/" dir is on a FAT partition on uc20 so the permissions are
+    # different here.
+    if ! is_core20_system; then
```
Comment (sergiocazzolato) [inline tests/main/install-store/task.yaml]: This is not executed on core systems, is it needed?

Comment (mvo5) [reply tests/main/install-store/task.yaml]: Aha, thank you! I missed that, will remove it again.

File: tests/main/docker-smoke/task.yaml [hunk 2]
```diff
@@ -1,7 +1,8 @@
 summary: Check that the the docker snap works basically
 
 # only run on ubuntus for now, the docker snap has issues on non-ubuntu ATM
-# TODO:UC20: enable for UC20
+# TODO:UC20: enable for UC20, fails with what looks like apparmor
+# errors trying to run kmod and open /run/system/sessions/1 and ptrace
```
Comment (anonymouse64) [inline tests/main/docker-smoke/task.yaml]: from the log that @mvo5 shared, the issue here is rather with overlayfs mounts, the /run/systemd/sessions/1 and ptrace denials are normal

File: tests/main/ubuntu-core-netplan/task.yaml [hunk 3]
```diff
@@ -1,5 +1,9 @@
 summary: Ensure that netplan works on Ubuntu Core with network-setup-{control,observe}
 
+#TODO:UC20: enable, fails right now with: "The permission of the
```
Comment (anonymouse64) [inline tests/main/ubuntu-core-netplan/task.yaml]: note that I think this is a more generic dbus activation issue on uc20

**Comment by mvo5** [comment]
Thanks Ian and Sergio!

**Pull request merged**